### PR TITLE
:seedling: Increase test coverage for finding outcomes

### PR DESCRIPTION
**Pull request by naveensrinivasan**
- Add tests for Outcome UnmarshalYAML function in `finding/finding_test.go`

#### What kind of change does this PR introduce?

(Is it a bug fix, feature, docs update, something else?)

- [ ] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the new behavior (if this is a feature change)?**

- [ ] Tests for the changes have been added (for bug fixes/features)

#### Does this PR introduce a user-facing change?

For user-facing changes, please add a concise, human-readable release note to
the `release-note`

(In particular, describe what changes users might need to make in their
application as a result of this pull request.)

```release-note

```

### CODE REVIEW

File: finding/finding_test.go [hunk 1]
```diff
@@ -198,3 +199,88 @@ func Test_FromBytes(t *testing.T) {
...
 	}
 }
+
+func TestOutcome_UnmarshalYAML(t *testing.T) {
+	type args struct {
+		n *yaml.Node
+	}
+	tests := []struct { //nolint:govet
+		name    string
+		o       Outcome
```
Comment (spencerschrock) [inline finding/finding_test.go:209]: wouldn't `o Outcome` be `want Outcome` in this case?

File: finding/finding_test.go [hunk 2]
```diff
@@ -198,3 +199,88 @@ func Test_FromBytes(t *testing.T) {
...
+			args: args{
+				n: &yaml.Node{
+					Kind:  yaml.ScalarNode,
+					Value: "Unknown",
+				},
+			},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
```
Comment (spencerschrock) [inline finding/finding_test.go:279]: Any reason not to parallelize? and therefore include a `tt := tt`?

File: finding/finding_test.go [hunk 3]
```diff
@@ -198,3 +199,88 @@ func Test_FromBytes(t *testing.T) {
...
+		t.Run(tt.name, func(t *testing.T) {
+			if err := tt.o.UnmarshalYAML(tt.args.n); (err != nil) != tt.wantErr {
```
Comment (spencerschrock) [inline finding/finding_test.go:281]: we don't want to unmarshall until the value we're looking for.
 
I think you need a local var here to unmarshall into, and then a comparison between that value and what we wanted?

right now we're only testing for an error.

**Review by spencerschrock — APPROVED** [review]

File: finding/finding_test.go (hunk 3 above)
Comment (naveensrinivasan) [reply finding/finding_test.go]: Agreed! Fixed it

File: finding/finding_test.go (hunk 2 above)
Comment (naveensrinivasan) [reply finding/finding_test.go:280]: 👍

File: finding/finding_test.go (hunk 1 above)
Comment (naveensrinivasan) [reply finding/finding_test.go]: 👍

**Pull request merged**